src/Provider/Yandex/Model/YandexAddress.php

```diff
@@ -0,0 +1,71 @@
+<?php
+
+/*
+ * This file is part of the Geocoder package.
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ *
+ * @license    MIT License
+ */
+
+namespace Geocoder\Provider\Yandex\Model;
+
+use Geocoder\Model\Address;
+
+/**
+ * @author Tobias Nyholm <tobias.nyholm@gmail.com>
+ */
+class YandexAddress extends Address
+{
+    /**
+     * @var string|null
+     */
+    private $precision;
+
+    /**
+     * The name of this location.
+     *
+     * @var string|null
+     */
+    private $name;
+
+    /**
+     * @return null|string
+     */
+    public function getPrecision()
+    {
+        return $this->precision;
+    }
+
+    /**
+     * @param null|string $precision
+     *
+     * @return YandexAddress
+     */
+    public function setPrecision($precision)
```

Comment:
The problem is that all providers is not setting all data. Some data might me permanently ignored by some providers. (`MaxMindBinary` is a perfect example). The `LocationBuilder` brings an easy way (with setter/typehints) to add some/all data. The logic (order of constructor parameters) is hidden away. Before the `LocationBuilder` was introduces we used an assoc array. 

Example:
```php
$builder->setE($httpResponse['e']);
$address = $builder->build();
```
Compared with: 
```php
$e = $httpResponse['e'];
$address = new Address(null, null, null, null, $e);
```

-------------

> But how to call the constructor to subclasses of Address with the LocationBuilder?